pliib.hpp

```diff
@@ -266,48 +266,48 @@ using namespace std;
         ret[rsz] = '\0';
         delete [] s;
         s = ret;
-    };
+    }
 
     inline void strip(std::string& s, char r){
         int slen = s.length();
         char* t = new char[slen + 1];
         memcpy(t, s.c_str(), slen);
         strip(t, slen, r);
         s = std::string(t);
-    };
+    }
 
     inline void slice(const char* s, size_t start, size_t end, char*& ret){
        strcopy(s + start, end - start, ret); 
-    };
+    }
 
 
     /** Removes a character from within a string **/
     inline void remove_char(char*& s, const int& len, char r){
         int write_index = 0;
         int read_index = 0;
-        for (int i = 0; i < len, read_index < len; ++i){
+        for (int i = 0; i < len && read_index < len; ++i){
             if (s[i] != r){
                 s[write_index] = s[read_index];
                 ++write_index;
             }
             ++read_index;
         }
         s[write_index] = '\0';
-    };
+    }
 
     /** Removes any null or whitespace characters from within a string
      *  where whitespace is ' ' or '\t' or '\n'.
      *  Complexity: linear time in |s| **/
     inline void remove_nulls_and_whitespace(char*& s, const int& len){
         size_t write_index = 0;
-        for (size_t i = 0; i < len; ++i){
+        for (int i = 0; i < len; ++i){
```

Comment:
Note to self: len is passed as int& here and had the extra promotion to size_t for no reason. I might consider making it a size_t in the future (to allow for huge strings).

Reply:
You may also want to consider pass `len` by value. It probably would be faster than using const reference. See: https://stackoverflow.com/a/3009568/357257